modules/core/shared/src/main/scala/io/circe/JsonObject.scala

```diff
@@ -166,6 +166,23 @@ sealed abstract class JsonObject extends Serializable {
    */
   final def filterKeys(pred: String => Boolean): JsonObject = filter(field => pred(field._1))
 
+  /**
+   * Perform a deep merge of this JSON object with another JSON object.
+   *
+   * Objects are merged by key, values from the argument JSON take
+   * precedence over values from this JSON. Nested objects are
+   * recursed.
+   *
+   * See [[Json.deepMerge]] for behavior of merging values that are not objects.
```

Comment:
We don't need to do it in this PR, but Scaladoc actually has a `@see` tag for cases like this.

Reply:
Ah nice, wasn't aware of that as I've never actually ended up building any scaladoc docs myself...